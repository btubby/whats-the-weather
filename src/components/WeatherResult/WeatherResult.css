.results {
    border-style: dotted;
    border-color: #812E7A;
    margin: 10px 0 10px 0;
    padding: 10px;
  }