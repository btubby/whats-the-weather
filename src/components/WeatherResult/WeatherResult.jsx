import React from "react";
import "./WeatherResult.css";

export const WeatherResult = ({searchItem, flatDisplay}) => {
  if (flatDisplay) {
    return (
      <div className="search-result-flat">
        {searchItem.name}, {searchItem.main.temp} &#x2103;, {searchItem.main.pressure} hpa
      </div>
    );
  } else {
    
    return (
      <div className="search-result">
       {searchItem.name}
       <br />
       {searchItem.main.temp} &#x2103;
       <br />
       {searchItem.main.pressure} hpa
       <br />
       {searchItem.main.humidity} %
      </div>
    );
   }
}