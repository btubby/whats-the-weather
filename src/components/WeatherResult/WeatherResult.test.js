import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import mockData from '../../data/weather.json';
import { WeatherResult } from './WeatherResult';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders", () => {
  act(() => {
    render(<WeatherResult response={mockData} />, container);
  });
  console.log(container.textContent)
  // test the component renders the city name, temp, pressure and humidity in the mock response
  expect(container.textContent.toBeDefined);
  expect(container.textContent).toContain('London');
  expect(container.textContent).toContain('285.59');
  expect(container.textContent).toContain('1032 hpa');
  expect(container.textContent).toContain('71 %');
  

});