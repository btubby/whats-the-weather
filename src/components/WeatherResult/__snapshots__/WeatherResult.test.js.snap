// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a WeatherResult component renders correctly 1`] = `
<div
  class="search-result"
>
  London
  <br />
  285.59
   
  ℃
  <br />
  1032
   hpa
  <br />
  71
   %
</div>
`;

exports[`Given a WeatherResult component with a searchHistory prop renders correctly 1`] = `
<div
  class="search-result-history"
>
  London
  <br />
  285.59
  ℃
  , 
  1032
  hpa, 
  71
  %
</div>
`;

exports[`Given a WeatherResult component, displaying temperature in Kelvin renders correctly 1`] = `
<div
  class="search-result"
>
  London
  <br />
  558.74
   
  K
  <br />
  1032
   hpa
  <br />
  71
   %
</div>
`;
